Makefile

```diff
@@ -9,7 +9,7 @@ TESTDIR = tests
 
 SPHINX_APIDOC = 
 
-BUILD_ORDER = ocrd_utils ocrd_models ocrd_modelfactory ocrd_validators ocrd
+BUILD_ORDER = ocrd_utils ocrd_models ocrd_modelfactory ocrd_validators ocrd_network ocrd
```

Comment:
I never had problems with the current build order (manually uninstalled and installed everything to check again). However, I like the suggestion. Future developments of `ocrd_network` may need that change.

Reply:
> (Resolver, only 1 place in ProcessingWorker for workspace_from_url; IMO we don't need that if we actually use the MongoDB Workspace entries to get the absolute path)

Right, we can just pass the full path to `run_processor()` and `run_cli()` and it will get resolved inside.

Reply:
> I never had problems with the current build order

You will only notice when you increase the version.



> What about the new CLIs? Doesn't the `ocrd` require `ocrd_network` installed before itself?

It does – see my comment above.



> > (Resolver, only 1 place in ProcessingWorker for workspace_from_url; IMO we don't need that if we actually use the MongoDB Workspace entries to get the absolute path)
> 
> Right, we can just pass the full path to `run_processor()` and `run_cli()` and it will get resolved inside. Seems we did not consider this and just took it the way it was implemented in #884.

Yes. As long as we can guarantee that the METS URL is a local path, available both to the Workspace Server and the Processing Server, then I believe nothing from the many different cases in `Resolver.workspace_from_url` is needed.

But that will not relieve us of the dependency on `ocrd`, because we would then need `Workspace` directly instead. And of course we always need `run_cli` and `run_processor` itself.

But perhaps avoiding circular dependency is not strictly necessary.

Reply:
Agreed.

I just tested with a new version number, and it did work. (Also with `pip install -e` BTW.)